sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -1131,6 +1144,37 @@ private void messageReceived(final ChannelHandlerContext context, final RntbdRes
         }
     }
 
+    private void handleGoneException(RxDocumentServiceRequest request, Exception exception) {
+
+        if (request.requestContext == null) {
+            return;
+        }
+
+        if (exception instanceof PartitionIsMigratingException) {
+            request.forceCollectionRoutingMapRefresh = true;
+            request.requestContext.forceRefreshAddressCache = true;
+        } else if (exception instanceof InvalidPartitionException) {
+            request.forceNameCacheRefresh = true;
+            request.requestContext.quorumSelectedLSN = -1;
+            request.requestContext.resolvedPartitionKeyRange = null;
+            request.requestContext.quorumSelectedStoreResponse = null;
+            request.requestContext.globalCommittedSelectedLSN = -1;
```

Comment:
why we are changing `globalCommittedSelectedLSN`  `quorumSelectedStoreResponse` etc?

Reply:
It doesn't have much meaning here since this request won't go through the consistency layer. But better to be safe.